pkg/logs/client/http/destination.go

```diff
@@ -56,13 +58,19 @@ func NewDestination(endpoint config.Endpoint, contentType string, destinationsCo
 // the error returned can be retryable and it is the responsibility of the callee to retry.
 func (d *Destination) Send(payload []byte) error {
 	ctx := d.destinationsContext.Context()
-	req, err := http.NewRequest("POST", d.url, strings.NewReader(string(payload)))
+
+	body, err := d.compression.compress(payload)
```

Comment:
Does not feel right to me to have a NoCompression which can compress, could we rename the abstraction to something more meaningful please, e.g. `BodyWriter`, `BodyBuilder` or a lambda `NewBody`: (bytes) -> (bytes, err)

Reply:
Indeed a better name/abstraction is `ContentType`